src/Util/ModuleUtils.php

```diff
@@ -74,18 +103,53 @@ public function downloadMainClass(string $moduleName, Version $version): void
 
         $response = $this->client->get(sprintf(self::GITHUB_MAIN_CLASS_ENDPOINT, $moduleName, $version->getTag(), $moduleName));
         file_put_contents($path . '/' . $moduleName . '.php', $response->getBody());
-        $this->saveDownloadUrl($moduleName, $version);
     }
 
-    /**
-     * @return Module[]
-     */
-    public function getLocalModules(): array
+    public function download(string $moduleName, Version $version): void
+    {
+        $path = join('/', [$this->moduleDir, $moduleName, $version->getTag()]);
+        if (!is_dir($path)) {
+            mkdir($path, 0777, true);
+        }
+
+        if ($version->getGithubUrl() === null) {
+            throw new RuntimeException(sprintf('Unable to download %s %s because it has no Github url', $moduleName, $version->getTag()));
+        }
+
+        $response = $this->client->get($version->getGithubUrl());
+        file_put_contents($path . '/' . $moduleName . '.zip', $response->getBody());
+    }
+
+    public function extractLogo(string $moduleName, Version $version): void
+    {
+        $path = join('/', [$this->moduleDir, $moduleName, $version->getTag(), $moduleName . '.zip']);
+        if (!file_exists($path)) {
+            return;
+        }
+
+        $moduleZip = new ZipArchive();
+        $moduleZip->open($path);
+        $icon = $moduleZip->getFromName($moduleName . '/logo.png');
+        $moduleZip->close();
+
+        file_put_contents(join('/', [$this->moduleDir, $moduleName, $version->getTag(), 'logo.png']), $icon);
```

Comment:
Same here, check the path

Be paranoid 😄

Reply:
The function returns if the path doesn't exists